Queueomatic/Server/Endpoints/Room/GetAll/GetAllRoomsEndpoint.cs

```diff
@@ -0,0 +1,35 @@
+﻿using FastEndpoints;
+using Queueomatic.DataAccess.UnitOfWork;
+using Queueomatic.Server.Services.HashIdService;
+using Queueomatic.Server.Services.RoomService;
+
+namespace Queueomatic.Server.Endpoints.Room.GetAll;
+
+public class GetAllRoomsEndpoint : Endpoint<GetAllRoomRequest, GetAllRoomsResponse>
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IRoomService _roomService;
+    private readonly IHashIdService _hashIdService;
+
+    public GetAllRoomsEndpoint(IUnitOfWork unitOfWork, IRoomService roomService, IHashIdService hashIdService)
+    {
+        _unitOfWork = unitOfWork;
+        _roomService = roomService;
+        _hashIdService = hashIdService;
+    }
+
+    public override void Configure()
+    {
+        Get("/rooms");
+        Roles("Administrator");
+    }
+
+    public override async Task HandleAsync(GetAllRoomRequest req, CancellationToken ct)
+    {
+        var roomId = _hashIdService.Decode(req.RoomId);
```

Comment:
What will be the response of the endpoint when roomId gets -1 as value?

Reply:
If you check the repository file you will see that the filter handles any negative numbers.

Reply:
Yes it is. For example if you decide to omit the query parameter and search only by name, you would want it to return those results.

Reply:
Okay. Makes sense.